src/Compilers/VisualBasic/Portable/Symbols/Metadata/PE/PEMethodSymbol.vb

```diff
@@ -1136,12 +1156,28 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols.Metadata.PE
             If Not _packedFlags.IsUseSiteDiagnosticPopulated Then
                 Dim errorInfo As DiagnosticInfo = CalculateUseSiteErrorInfo()
                 EnsureTypeParametersAreLoaded(errorInfo)
+                CheckUnmanagedCallersOnly(errorInfo)
                 Return InitializeUseSiteErrorInfo(errorInfo)
             End If
 
             Return _uncommonFields?._lazyUseSiteErrorInfo
         End Function
 
+        Private Sub CheckUnmanagedCallersOnly(ByRef errorInfo As DiagnosticInfo)
+            If Not _packedFlags.IsUnmanagedCallersOnlyPopulated Then
+                Dim hasUnmanagedCallersOnly As Boolean =
+                    DirectCast(ContainingModule, PEModuleSymbol).Module.FindLastTargetAttribute(_handle, AttributeDescription.UnmanagedCallersOnlyAttribute).HasValue
```

Comment:
>FindLastTargetAttribute [](start = 72, length = 23)

Do we actually need to find the last application or any would be sufficient for our purpose?

Reply:
> Do we actually need to find the last application or any would be sufficient for our purpose?

Any would be sufficient. I'll make the find first target attribute helper internal, it's private today.

---
In reply to: [566918784](https://github.com/dotnet/roslyn/pull/50886#discussion_r566918784) [](ancestors = 566918784)